cpp/src/arrow/dataset/test_util.h

```diff
@@ -139,5 +139,45 @@ std::unique_ptr<GeneratedRecordBatch<Gen>> MakeGeneratedRecordBatch(
   return internal::make_unique<GeneratedRecordBatch<Gen>>(schema, std::forward<Gen>(gen));
 }
 
+/// \brief A dummy FileFormat implementation
+class DummyFileFormat : public FileFormat {
+ public:
+  std::string name() const override { return "dummy"; }
+
+  /// \brief Return true if the given file extension
+  bool IsKnownExtension(const std::string& ext) const override { return ext == name(); }
+
+  /// \brief Open a file for scanning (always returns an empty iterator)
+  Status ScanFile(const FileSource& source, std::shared_ptr<ScanOptions> scan_options,
+                  std::shared_ptr<ScanContext> scan_context,
+                  std::unique_ptr<ScanTaskIterator>* out) const override {
+    *out = internal::make_unique<EmptyIterator<std::unique_ptr<ScanTask>>>();
```

Comment:
Have we fixed `make_unique` for MSVC yet?

Reply:
I still don't know what that was about. The version of MSVC which was complaining in CI is older than any I have access to, and I can't reproduce it.

Reply:
https://issues.apache.org/jira/browse/ARROW-5121